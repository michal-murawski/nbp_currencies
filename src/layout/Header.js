import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Typography from 'material-ui/Typography';
import AppBar from 'material-ui/AppBar';
import { HeaderStyles } from './styles';

export const headerTitle = 'Pick your favourite currencies! :)';

const Header = props => {
  const { classes, children } = props;

  return (
    <AppBar position="static" color="default">
      <Typography variant="display3" className={classes.title}>
        {headerTitle}
      </Typography>
      {children}
    </AppBar>
  );
};

Header.propTypes = {
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
    .isRequired,
  classes: PropTypes.object.isRequired,
};

export const HeaderRaw = Header;

export default withStyles(HeaderStyles)(Header);
