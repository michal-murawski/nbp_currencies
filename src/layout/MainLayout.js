import React from 'react';
import PropTypes from 'prop-types';
import Navigation from './navigation/Navigation';
import Header from './Header';
import Content from './Content';

const MainLayout = props => {
  return (
    <div>
      <Header>
        <Navigation />
      </Header>
      <Content>{props.children}</Content>
    </div>
  );
};

MainLayout.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
};

export const MainLayouRaw = MainLayout;

export default MainLayout;
