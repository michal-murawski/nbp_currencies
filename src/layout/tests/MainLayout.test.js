import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Header from '../Header';

import { MainLayouRaw } from '../MainLayout';

const TestChild = () => <div>Test child text</div>;
const mockClasses = { title: 'title_class' };


describe('layout/MainLayouRaw', () => {
  const wrapper = shallow(
    <MainLayouRaw classes={mockClasses}>
      <TestChild />
      <TestChild />
      <TestChild />
    </MainLayouRaw>
  );

  it('should render properly the MainLayouRaw', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render Header', () => {
    expect(wrapper.find(Header)).toHaveLength(1);
  });
});
