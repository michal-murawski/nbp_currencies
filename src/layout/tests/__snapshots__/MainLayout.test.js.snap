// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout/MainLayouRaw should render properly the MainLayouRaw 1`] = `
<div>
  <WithStyles(Header)>
    <Navigation />
  </WithStyles(Header)>
  <WithStyles(Content)>
    <TestChild />
    <TestChild />
    <TestChild />
  </WithStyles(Content)>
  <Connect(Snackbar) />
</div>
`;
